core/src/main/java/com/graphhopper/storage/index/LocationIndexTree.java

```diff
@@ -564,6 +572,133 @@ protected boolean check(int node, double normedDist, int wayIndex, EdgeIteratorS
         return closestMatch;
     }
 
+    /**
+     * Returns all edges that are within the specified radius around the queried position.
+     *
+     * @param queryLat latitude to search from
+     * @param queryLon longitude to search from
+     * @param edgeFilter only include edges from edgeFilter
+     * @param radius include all results within this radius (in meters) - assuming it's not
+     *        bounded by maxIterations.
+     * @param maxIterations if not null, search only a square of (2 * maxIterations - 1) tiles
+     *        wide/high (with center tile containing (queryLat, queryLon)). Note that it is
+     *        possible that the search radius exceeds this square - in this case, an exception
+     *        will be thrown.
+     */
+    public List<QueryResult> findWithinRadius(final double queryLat, final double queryLon,
+            final EdgeFilter edgeFilter, final double radius, final Integer maxIterations) {
+
+        // check the max iterations:
+        int requiredIterationsForThisRadius = (int) Math.ceil(radius / this.minResolutionInMeter)
+                + 1; // note that this is a conservative upper bound
+        int _maxIterations;
```

Comment:
Not sure, as it's not really tmp. I would favor removing maxIterations from the parameter list :). But let's discuss this in the comment above.